homeassistant/components/lifx/sensor.py

```diff
@@ -37,6 +46,46 @@ async def async_setup_entry(
     coordinator: LIFXUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
     async_add_entities([LIFXRssiSensor(coordinator.sensor_coordinator, RSSI_SENSOR)])
 
+    if lifx_features(coordinator.device)["multizone"]:
+        async_add_entities(
+            [LIFXZonesSensor(coordinator.sensor_coordinator, ZONES_SENSOR)]
+        )
+
+
+class LIFXZonesSensor(LIFXSensorEntity, SensorEntity):
+    """LIFX Zones sensor for linear multizone devices."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: LIFXSensorUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialise the sensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._attr_name = description.name
+        self._attr_unique_id = f"{coordinator.parent.serial_number}_{description.key}"
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._async_update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Handle coordinator updates."""
+        self._attr_native_value = self.coordinator.zones_count
+        self._attr_extra_state_attributes = self.coordinator.zone_colors
```

Comment:
Each zone should be its own sensor entity instead.  [We shouldn't put all the attributes for every zone into a single sensor entity](https://github.com/home-assistant/core/pull/84934/files#diff-69691db0e3fa99416b32d19456ed5ab9fdd89264faba240b9e863017ad63fc81R83).  

Ideally we mirror what `wled` did with segments and each zone is its own light entity.

https://github.com/home-assistant/core/blob/3c70dd9b425fdab979795c68bd86f6bff4b4e8f7/homeassistant/components/wled/light.py#L102

Reply:
Except the WLED implementation isn't feasible because you're unlikely to have as many segments as you would zones on a LIFX Z or Beam. All of mine are >60 zones; most are 80. I really don't want 80 light entities in a single device.

Reply:
I'll ask to see if someone else has another idea on how to represent this.

Reply:
The "one sensor per zone" option doesn't work either because there are actually four sensors per zone, i.e. hue, saturation, brightness and kelvin. But there isn't quite enough to make a light entity, because I really don't want folks to control individual zones like that (the protocol just isn't efficient for it).

Reply:
A picture is worth 1000 words:

![Screenshot 2023-03-07 at 16-02-20 Settings – Home Assistant](https://user-images.githubusercontent.com/103232/223325372-223bc1b3-4133-4827-ba4b-5200dd0268f7.png)

I have another 40 zones on this Beam and it's my shortest one. And if I make the value any longer (like adding legends) then it truncates the name of the entity, so it becomes impossible to see which zone is which. 

Next I'll try creating them as light entities that just don't do anything to see what it looks like.

Reply:
Having them as light entities is _prettier_ but the user experience isn't that much better:
![Screenshot 2023-03-07 at 16-44-35 Settings – Home Assistant](https://user-images.githubusercontent.com/103232/223331480-ba0551d7-c608-40a9-a8e7-f1164954bb10.png)

Keep in mind there are another **37** light entities on this page.

Reply:
This is my favourite screenshot. It's the auto-generated dashbord at 4k with three screens worth of more entities hidden below the fold:

![screenshot-localhost_8123-2023 03 07-16_51_18](https://user-images.githubusercontent.com/103232/223332581-f5dda423-ad02-4a4c-913c-bf5ba0c1be60.png)

Those are just my linear multizone devices. I haven't yet added my LIFX Tiles, one of which has 256 zones and the other has 320 zones.

We need additional attributes that are not recorded in the state machine/recorder. :)

Reply:
@bdraco @frenck I've been thinking about this and comparing it with the WLED Segments functionality and it occurred to me that Home Assistant could probably use a Color entity, which could be represented by any of the color modes, i.e. RGB, HSBK, XY, even Pantone. 

Thoughts? I'd prefer to know at least vaguely whether this is a good idea before doing the work required to pitch a new entity type.